lib/app/hive.dart

```diff
@@ -33,9 +33,7 @@ extension SaveableEntity<E extends Entity<E>> on Entity<E> {
 /// different types.
 @immutable
 class Id<E extends Entity<E>> {
-  Id(this.value)
-      : assert(value != null),
-        assert(value.isNotEmpty);
+  const Id(this.value) : assert(value != null);
```

Comment:
Why would we now allow empty values?

Reply:
Because the `FilePath`'s `copyWith` method should support `null` as `parentId`, we use a default value `_defaultFile`. Because default values of optional parameters must be `const`, `Id`'s constructor has to be `const` which means `isNotEmpty` doesn't work anymore 😞